printing/printer.ml

```diff
@@ -442,7 +442,13 @@ let pr_predicate pr_elt (b, elts) =
     else
       if List.is_empty elts then str"none" else pr_elts
 
-let pr_cpred p = pr_predicate (pr_constant (Global.env())) (Cpred.elements p)
+let pr_cpred p =
+  let safe_pr_constant env kn =
+    try pr_constant env kn
+    with Not_found when !Flags.in_debugger || !Flags.in_toplevel ->
```

Comment:
I don't think it's a good idea in the long run to protect the printing of data in the wrong environment by this kind of flags. Instead, we should do this unconditionally.

Reply:
Unconditionally means that a user will potentially see debugging data (e.g. internal module names) in place of an anomaly, right? Indeed, that's debatable.

In any case, I'm following what is done in other places. If a change of strategy is planned, it should probably be global.

Reply:
Yes, I agree this should be both pondered and global if eventually performed.